ckanext/datapreview/lib/helpers.py

```diff
@@ -197,18 +208,24 @@ def proxy_query(resource, url, query):
 
     length = query.get('length',
                        get_resource_length(url, resource,
-                                           trans.requires_size_limit))
+                                           trans.requires_size_limit()))
 
     log.debug('The file at %s has length %s', url, length)
 
     max_length = int(query['size_limit'])
 
-    if query.get('archived', True):
-        log.debug("Skipping size check when reading from archive")
+    transformer_requires_size_limit = trans.requires_size_limit()
+    log.debug('Size=%s Archived=%s Transformer-limited=%s Limit=%s',
+              int(length), query.get('archived'),
+              transformer_requires_size_limit, max_length)
+    if query.get('archived', True) and not transformer_requires_size_limit:
+        log.debug('Skipping size check - reading from archive and the '
+                  'transformer for this format does not require a limit')
     else:
-        # We only do the length check when we are working with remote files as
-        # they might take too long to download.
-        if length and trans.requires_size_limit and int(length) > max_length:
+        # Do size check
+        #  - remote files may take too long to download
+        #  - some file formats need loading fully into memory and take too much
+        if length and int(length) > max_length:
```

Comment:
This looks fine. I think requires_size_limit is always True to happy for it to be removed.